# DATA UNDERSTANDING
In this section, we are going introduce ourselves to the data we are going to use, exploring them for insights and understanding before we begin our analysis. This section will include data description and summary to know what we are getting into.

In this repository, under the file path `data/raw/zippedData`, we have 6 zipped datasets namely `im.db`, `bom.movie_gross.csv`, `tn.movie_budgets.csv`, `rt.reviews.tsv`, `rt.movie_info.tsv`, and `tmdb.movies.csv`.
We are going to explore them one by one focusing on their sources, structures, formats, why they are suitable and if they are in the best shape for analysis.

The analysis tools we will be using in this project include;
- Pandas

- SQLite3

- Numpy

- Seaborn

- Matplotlib

In [ ]:
# Importing the necessary libraries for analysis.

import pandas as pd
import numpy as np
import sqlite3
import string as str
import seaborn as sns
import statsmodels.api as sm
import matplotlib.pyplot as plt

## rt.reviews.tsv
### Data source
The `rt.reviews.tsv.gz` dataset is sourced from the Rotten Tomatoes website, a film and television review aggregator and user community. The data contains movie reviews and ratings from various movie critics dating from the year 2000 to 2018.

### Why is the data suitable for the project?
* The data contains movie reviews, ratings, user comments, and critical reception, which help predict box office performance and overall success. This information also reveals audience preferences for genres, themes, and storylines, which is important to help studios tailor their content to meet market demand and avoid oversaturation in certain areas.

* User reviews and comments provide valuable insights into audience perceptions of specific films and the overall cinematic landscape. Most importantly, this information is crucial for attracting investors and making informed decisions about green-lighting projects.

In [2]:
# We are reading the data into the 'rt_reviews, variable using the .read_csv() pandas function to be able to access the data in form of a dataframe

rt_reviews = pd.read_csv("data/raw/zippedData/rt.reviews.tsv.gz", compression= 'gzip', delimiter= '\t', encoding= 'latin-1', index_col= False)
rt_reviews.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [3]:
# This code will print out the shape and the data's columns.

print(f'The data has a {rt_reviews.shape} structure.')
print('`rt_reviews` has 8 columns namely:', rt_reviews.columns)

The data has a (54432, 8) structure.
`rt_reviews` has 8 columns namely: Index(['id', 'review', 'rating', 'fresh', 'critic', 'top_critic', 'publisher',
       'date'],
      dtype='object')


In [4]:
# We will be using the .info() function to explore the data further.
# this function will show us the columns' datatypes, count of non null values, allowing us to identify columns with null values.

rt_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [5]:
# Other than the .info() function, .describe() function also gives a summary of the dataset.
# This function gives us the statistical summary of the data.

rt_reviews.describe(include= 'all')

id                   review rating  fresh        critic  \
count   54432.000000                    48869  40915  54432         51710   
unique           NaN                    48682    186      2          3496   
top              NaN  Parental Content Review    3/5  fresh  Emanuel Levy   
freq             NaN                       24   4327  33035           595   
mean     1045.706882                      NaN    NaN    NaN           NaN   
std       586.657046                      NaN    NaN    NaN           NaN   
min         3.000000                      NaN    NaN    NaN           NaN   
25%       542.000000                      NaN    NaN    NaN           NaN   
50%      1083.000000                      NaN    NaN    NaN           NaN   
75%      1541.000000                      NaN    NaN    NaN           NaN   
max      2000.000000                      NaN    NaN    NaN           NaN   

          top_critic        publisher             date  
count   54432.000000            54123            54432  
unique           NaN             1281             5963  
top              NaN  eFilmCritic.com  January 1, 2000  
freq             NaN              673             4303  
mean        0.240594              NaN              NaN  
std         0.427448              NaN              NaN  
min         0.000000              NaN              NaN  
25%         0.000000              NaN              NaN  
50%         0.000000              NaN              NaN  
75%         0.000000              NaN              NaN  
max         1.000000              NaN              NaN

In [6]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = rt_reviews.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

rating       13517
review        5563
critic        2722
publisher      309
dtype: int64

In [7]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = rt_reviews.duplicated().sum()
duplicates

9

### Data summary
* The dataset contains structured data stored in a TSV (tab-separated values) file containing both non-categorical and categorical data. 

* The dataset contains data in 54432 rows and 8 columns namely, 'id', 'review', 'rating', 'fresh', 'critic', 'top_critic', 'publisher', and 'date'. All the columns are well labeled and easy to interpret.

* Out of the 8 columns, only 4 contain null values ranging from as low as 300 to 13,517 null values. The dataset also contains 9 duplicated rows.

* The dataset contains mostly object datatypes, hence there is no quantitative analysis needed.

### Data Quality Issues
- Incompleteness - 4 columns contain missing values. Missing values in data can lead to inaccuracies and misleading results, which subsequently leads to inaccurate insights.

- Duplicate records - The dataset contains 9 duplicated records, which could skew the results as certain records provide the same information twice.

### Next steps
The data is mostly clean and has a lot of information to offer, however, a few columns need some cleaning. The cleaning involves;

* Handle missing and duplicated records.

## rt.movies_info.tsv
### Data source
The `rt.movie_info.tsv` data is extracted from Rotten Tomatoes website and contains information about movies and television series including their genres, ratings, synopsis as well as the movie's director, writer, release date and run-time.

### Why is the data suitable for the project?
* This data allows us to analyze the impact of writers and directors on previous films' success providing valuable information for casting decisions. Additionally, real-time review data can track a film's performance after its release, allowing studios to adapt marketing strategies and address potential issues. 

* Comparing a film's performance against similar titles in terms of review scores and box office success allows studios to identify areas for improvement and best practices.


In [8]:
# We are reading the data into the 'rt_movies, variable using the .read_csv() pandas function to be able to access the data in form of a dataframe

rt_movies = pd.read_csv("data/raw/zippedData/rt.movie_info.tsv.gz", compression= 'gzip', delimiter= '\t', encoding= 'latin-1', index_col= False)
rt_movies.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [9]:
# This code will print out the shape and the data's columns.

print(f'The data has a {rt_movies.shape} structure.')
print('`rt_reviews` has 8 columns namely:', rt_movies.columns)

The data has a (1560, 12) structure.
`rt_reviews` has 8 columns namely: Index(['id', 'synopsis', 'rating', 'genre', 'director', 'writer',
       'theater_date', 'dvd_date', 'currency', 'box_office', 'runtime',
       'studio'],
      dtype='object')


In [10]:
# We will be using the .info() function to explore the data further.
# this function will show us the columns' datatypes, count of non null values, allowing us to identify columns with null values.

rt_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [11]:
# Other than the .info() function, .describe() function also gives a summary of the dataset.
# This function gives us the statistical summary of the data.

rt_movies.describe(include= 'all')

id                                           synopsis rating  \
count   1560.000000                                               1498   1557   
unique          NaN                                               1497      6   
top             NaN  A group of air crash survivors are stranded in...      R   
freq            NaN                                                  2    521   
mean    1007.303846                                                NaN    NaN   
std      579.164527                                                NaN    NaN   
min        1.000000                                                NaN    NaN   
25%      504.750000                                                NaN    NaN   
50%     1007.500000                                                NaN    NaN   
75%     1503.250000                                                NaN    NaN   
max     2000.000000                                                NaN    NaN   

        genre          director       writer theater_date     dvd_date  \
count    1552              1361         1111         1201         1201   
unique    299              1125         1069         1025          717   
top     Drama  Steven Spielberg  Woody Allen  Jan 1, 1987  Jun 1, 2004   
freq      151                10            4            8           11   
mean      NaN               NaN          NaN          NaN          NaN   
std       NaN               NaN          NaN          NaN          NaN   
min       NaN               NaN          NaN          NaN          NaN   
25%       NaN               NaN          NaN          NaN          NaN   
50%       NaN               NaN          NaN          NaN          NaN   
75%       NaN               NaN          NaN          NaN          NaN   
max       NaN               NaN          NaN          NaN          NaN   

       currency  box_office     runtime              studio  
count       340         340        1530                 494  
unique        1         336         142                 200  
top           $  32,000,000  90 minutes  Universal Pictures  
freq        340           2          72                  35  
mean        NaN         NaN         NaN                 NaN  
std         NaN         NaN         NaN                 NaN  
min         NaN         NaN         NaN                 NaN  
25%         NaN         NaN         NaN                 NaN  
50%         NaN         NaN         NaN                 NaN  
75%         NaN         NaN         NaN                 NaN  
max         NaN         NaN         NaN                 NaN

In [12]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = rt_movies.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

box_office      1220
currency        1220
studio          1066
writer           449
dvd_date         359
theater_date     359
director         199
synopsis          62
runtime           30
genre              8
rating             3
dtype: int64

In [13]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = rt_movies.duplicated().sum()
duplicates

0

### Data summary
*  The dataset contains structured data stored in a TSV file containing both numerical and categorical data.

* The dataset contains data in 1560 rows and 12 columns namely, 'id', 'synopsis', 'rating', 'genre', 'director', 'writer', 'theater_date', 'dvd_date', 'currency', 'box_office', 'runtime' and 'studio'. All the columns are well labeled and easy to interpret.

* Out of the 12 columns, 11 contain null values ranging from as low as 3 to 1220 null values. The dataset also contains 0 duplicated rows.

* The dataset contains object datatypes, hence there is no quantitative analysis needed.

### Data Quality Issues
- Incompleteness - 11 columns contain missing values. Missing values in data can lead to inaccuracies and misleading results, which subsequently leads to inaccurate insights.

- Inaccuracy - Several columns are in the wrong data types, hence resulting in inaccurate data. inaccurate information can cause significant problems with severe consequences. 

### Next steps
The data is mostly clean and has a lot of information to offer, however, a few columns need some cleaning. The cleaning involves;

* Handle missing.

* Data type conversion.

## tmdb.movies.csv
### Data source
The `tmdb.movies.csv` data is extracted from TheMovieDB, a community built movie and TV database. It contains information on the popularity, genres and movie ratings.

### Why the data is suitable for the project
* This dataset is suitable for the project because it allows analysis of past movie performance data, including audience ratings, and box office numbers, to create models that predict the potential success of upcoming films. Also, community reviews and ratings reveal audience tastes and preferences, which can help studios tailor their content to appeal to specific demographics and genres. 

* By understanding audience preferences and geographical trends, studios can tailor their distribution strategies, including release dates, theater selection, and marketing efforts. It facilitates collaboration between stakeholders by providing a shared platform for accessing information and discussing potential projects. 

* The database provides a wealth of data for market research, allowing studios to understand trends, competitor activities, and potential opportunities.

In [14]:
# We are reading the data into the 'tmdb_movies' variable using the .read_csv() pandas function to be able to access the data in form of a dataframe

tmdb_movies = pd.read_csv("data/raw/zippedData/tmdb.movies.csv.gz", compression= 'gzip', delimiter= ',', encoding= 'latin-1', index_col= False)
tmdb_movies.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [15]:
# This code will print out the shape and the data's columns.

print(f'The data has a {tmdb_movies.shape} structure.')
print('`tmdb_movies` has 10 columns namely:', tmdb_movies.columns)

The data has a (26517, 10) structure.
`tmdb_movies` has 10 columns namely: Index(['Unnamed: 0', 'genre_ids', 'id', 'original_language', 'original_title',
       'popularity', 'release_date', 'title', 'vote_average', 'vote_count'],
      dtype='object')


In [16]:
# We will be using the .info() function to explore the data further.
# this function will show us the columns' datatypes, count of non null values, allowing us to identify columns with null values.

tmdb_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


In [17]:
# Other than the .info() function, .describe() function also gives a summary of the dataset.
# This function gives us the statistical summary of the data.

tmdb_movies.describe(include= 'all')

Unnamed: 0 genre_ids             id original_language original_title  \
count   26517.00000     26517   26517.000000             26517          26517   
unique          NaN      2477            NaN                76          24835   
top             NaN      [99]            NaN                en           Eden   
freq            NaN      3700            NaN             23291              7   
mean    13258.00000       NaN  295050.153260               NaN            NaN   
std      7654.94288       NaN  153661.615648               NaN            NaN   
min         0.00000       NaN      27.000000               NaN            NaN   
25%      6629.00000       NaN  157851.000000               NaN            NaN   
50%     13258.00000       NaN  309581.000000               NaN            NaN   
75%     19887.00000       NaN  419542.000000               NaN            NaN   
max     26516.00000       NaN  608444.000000               NaN            NaN   

          popularity release_date  title  vote_average    vote_count  
count   26517.000000        26517  26517  26517.000000  26517.000000  
unique           NaN         3433  24688           NaN           NaN  
top              NaN   2010-01-01   Home           NaN           NaN  
freq             NaN          269      7           NaN           NaN  
mean        3.130912          NaN    NaN      5.991281    194.224837  
std         4.355229          NaN    NaN      1.852946    960.961095  
min         0.600000          NaN    NaN      0.000000      1.000000  
25%         0.600000          NaN    NaN      5.000000      2.000000  
50%         1.374000          NaN    NaN      6.000000      5.000000  
75%         3.694000          NaN    NaN      7.000000     28.000000  
max        80.773000          NaN    NaN     10.000000  22186.000000

In [18]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = tmdb_movies.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

Series([], dtype: int64)

In [19]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = tmdb_movies.duplicated().sum()
duplicates

0

### Data summary
* The data is stored in a CSV file containing both non-categorical and categorical data.

* The dataset has 26517 rows and 9 columns namely, 'genre_ids', 'id', 'original_language', 'original_title', 'popularity', 'release_date', 'title', 'vote_average', and 'vote_count'.

* It has no duplicate records and also no missing records.

### Data Quality
The dataset is mostly clean, with no apparent issues.

## tn.movie_budget.csv
### Data source
The `tn.movie_budgets.csv` dataset is extracted from 'The Numbers' box office revenue tracking website. The data contains the production budget, domestic gross as well as international gross of movies, providing a performance analysis of the movies since their release dates.

### Why the data is suitable for the project
* Data from a film revenue tracker website is crucial for movie studio stakeholders as it provides insights into a film's financial performance, helping with decision-making and planning throughout the film production process. This data enables better investment decisions, more effective marketing strategies, and improved content planning. 

* Analyzing past performance data, including box office sales, streaming numbers, and merchandise sales, allows studios to predict the potential profitability of upcoming films. This helps in securing investment and making informed decisions about production budgets and marketing campaigns. 

* Revenue trackers provide a clear picture of the financial risks and potential rewards associated with different film projects, allowing studios to allocate resources more effectively.

In [20]:
# We are reading the data into the 'movie_budgets' variable using the .read_csv() pandas function to be able to access the data in form of a dataframe

movie_budgets = pd.read_csv("data/raw/zippedData/tn.movie_budgets.csv.gz", compression= 'gzip', delimiter= ',', encoding= 'latin-1', index_col= False)
movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [21]:
# This code will print out the shape and the data's columns.

print(f'The data has a {movie_budgets.shape} structure.')
print('`movie_budgets` has 6 columns namely:', movie_budgets.columns)

The data has a (5782, 6) structure.
`movie_budgets` has 6 columns namely: Index(['id', 'release_date', 'movie', 'production_budget', 'domestic_gross',
       'worldwide_gross'],
      dtype='object')


In [22]:
# We will be using the .info() function to explore the data further.
# this function will show us the columns' datatypes, count of non null values, allowing us to identify columns with null values.

movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [23]:
# Other than the .info() function, .describe() function also gives a summary of the dataset.
# This function gives us the statistical summary of the data.

movie_budgets.describe(include= 'all')

id  release_date movie production_budget domestic_gross  \
count   5782.000000          5782  5782              5782           5782   
unique          NaN          2418  5698               509           5164   
top             NaN  Dec 31, 2014  Home       $20,000,000             $0   
freq            NaN            24     3               231            548   
mean      50.372363           NaN   NaN               NaN            NaN   
std       28.821076           NaN   NaN               NaN            NaN   
min        1.000000           NaN   NaN               NaN            NaN   
25%       25.000000           NaN   NaN               NaN            NaN   
50%       50.000000           NaN   NaN               NaN            NaN   
75%       75.000000           NaN   NaN               NaN            NaN   
max      100.000000           NaN   NaN               NaN            NaN   

       worldwide_gross  
count             5782  
unique            5356  
top                 $0  
freq               367  
mean               NaN  
std                NaN  
min                NaN  
25%                NaN  
50%                NaN  
75%                NaN  
max                NaN

In [24]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = movie_budgets.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

Series([], dtype: int64)

In [25]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = movie_budgets.duplicated().sum()
duplicates

0

### Data summary
* The dataset contains structured data stored in a CSV file containing numerical data.

* The file contains 5782 rows and 6 columns namely, 'id', 'release_date', 'movie', 'production_budget', 'domestic_gross', and 'worldwide_gross' all well labeled and interpretable.

* According to the data, the most recent movie is Moonfall which was produced on 2020-12-31, while the oldest movie to be produced is The Birth of a Nation produced on 1915-02-08.

* The  movie with the highest production budget is Avatar With a budget allocation of $425,000,000, whereas My Date With Drew had the lowest budget standing at $1100

* The movie with the highest domestic income was Star Wars Ep. VII: The Force Awakens with an average amount of $760,507,625. The film with the highest worldwide gross was Avatar with an estimated return of $2,776,345,279.

### Data quality issues
- Inaccuracies

### Next steps
As noted from the dataset, columns' datatypes need to be formated as follows: 

 1. release_date needs to be formated into a date time object 

 2. production_budget needs to be formated into an int object and the dollar signs and commas removed

 3. domestic_gross needs to be formated into an int object and the dollar signs and commas removed
 
 4. worldwide_gross needs to be formated into and int object and the dollar signs and commas removed

## bom.movie_gross
### Data source
The `bom.movie_gross.csv` dataset is sourced from the 'Box Office Mojo' (BOM) reporting and analysis service website. The dataset contains systematic and algorithmic domestic and international box office revenues.

### Why the data is suitable for the project
* Analyzing gross data can help studios identify areas where they can streamline processes and optimize resource allocation. For example, data on production costs, marketing expenses, and distribution channels can be used to identify opportunities for cost savings and efficiency improvements. 

* By monitoring the returns on different film projects, studios can track their overall profitability and make adjustments to their production and marketing strategies as needed. 

* Data from revenue trackers allows studios to compare their performance against that of other films and companies, providing insights into their competitive standing and areas for improvement.

* The data empowers stakeholders to make better decisions, reduce risks, and optimize their chances of success in the competitive film industry. 

* By tracking the performance of different marketing channels and release strategies, studios can optimize their efforts to maximize reach and revenue. 

In [26]:
# We are reading the data into the 'movie_gross' variable using the .read_csv() pandas function to be able to access the data in form of a dataframe

movie_gross = pd.read_csv("data/raw/zippedData/bom.movie_gross.csv.gz", compression= 'gzip', delimiter= ',', encoding= 'latin-1', index_col= False)
movie_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [27]:
# This code will print out the shape and the data's columns.

print(f'The data has a {movie_gross.shape} structure.')
print('`movie_gross` has 5 columns namely:', movie_gross.columns)

The data has a (3387, 5) structure.
`movie_gross` has 5 columns namely: Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')


In [28]:
# We will be using the .info() function to explore the data further.
# this function will show us the columns' datatypes, count of non null values, allowing us to identify columns with null values.

movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [29]:
# Other than the .info() function, .describe() function also gives a summary of the dataset.
# This function gives us the statistical summary of the data.

movie_gross.describe(include= 'all')

title studio  domestic_gross foreign_gross         year
count        3387   3382    3.359000e+03          2037  3387.000000
unique       3386    257             NaN          1204          NaN
top     Bluebeard    IFC             NaN       1200000          NaN
freq            2    166             NaN            23          NaN
mean          NaN    NaN    2.874585e+07           NaN  2013.958075
std           NaN    NaN    6.698250e+07           NaN     2.478141
min           NaN    NaN    1.000000e+02           NaN  2010.000000
25%           NaN    NaN    1.200000e+05           NaN  2012.000000
50%           NaN    NaN    1.400000e+06           NaN  2014.000000
75%           NaN    NaN    2.790000e+07           NaN  2016.000000
max           NaN    NaN    9.367000e+08           NaN  2018.000000

In [30]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = movie_gross.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

foreign_gross     1350
domestic_gross      28
studio               5
dtype: int64

In [31]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = movie_gross.duplicated().sum()
duplicates

0

### Data summary
* The dataset contains structured data stored in a CSV file containing numerical and non_categorical data.

* The file contains 3387 rows and 5 columns namely, 'title', 'studio', 'domestic_gross', 'foreign_gross', and 'year' all well labeled and interpretable.

* Out of the 5 columns, 3 contain null values and the numerical values are stored as objects.

### Data quality issues
- Incompleteness - 11 columns contain missing values. Missing values in data can lead to inaccuracies and misleading results, which subsequently leads to inaccurate insights.

- Inaccuracy - Several columns are in the wrong data types, hence resulting in inaccurate data. inaccurate information can cause significant problems with severe consequences. 

### Next steps
- Data cleaning

- Data conversion

## im.db database
### Data source
The `im.db` data is extracted from the 'Internet Movie Database' (IMDB). The database contains information relating to films, TV series, production crew and their biographies, as well as movie ratings.

### Why the data is suitable for the project
Data from a film database websites provide valuable insights for movie studio stakeholders, enabling them to make more informed decisions across various aspects of film production and distribution. This data helps with everything from identifying emerging trends and predicting box office success to guiding script development and optimizing marketing campaigns. 

* Historical data on previous films, including ratings, reviews, and box office performance, can be used to predict the potential success of new releases. Understanding which factors, such as actor popularity, director reputation, and genre trends, positively influence box office results. 

* Data can provide insights into audience preferences and expectations, which can help inform the development of compelling and engaging scripts. Studios can use data to identify successful narrative structures, character archetypes, and thematic elements that resonate with audiences. 

* Data can be used to target specific demographics and tailor marketing messages to resonate with different audience segments. Studios can also optimize showtime schedules and distribution strategies based on past performance data and predictions. 

* Studios can use data to personalize recommendations and create more engaging user experiences on their streaming platforms and websites. This can lead to increased user engagement and satisfaction, ultimately driving revenue

*  The database can help studios identify promising actors, directors, and other talent by tracking their past performances and reviews. 

In [32]:
# Connecting the database and reading it into the `conn` variable 
conn = sqlite3.connect('data/raw/im.db')

# In this code we are reading the tables into the table variable to get the lay of the database
table = pd.read_sql("""SELECT *
                      FROM sqlite_master""", conn)
table

type           name       tbl_name  rootpage  \
0  table   movie_basics   movie_basics         2   
1  table      directors      directors         3   
2  table      known_for      known_for         4   
3  table     movie_akas     movie_akas         5   
4  table  movie_ratings  movie_ratings         6   
5  table        persons        persons         7   
6  table     principals     principals         8   
7  table        writers        writers         9   

                                                 sql  
0  CREATE TABLE "movie_basics" (\n"movie_id" TEXT...  
1  CREATE TABLE "directors" (\n"movie_id" TEXT,\n...  
2  CREATE TABLE "known_for" (\n"person_id" TEXT,\...  
3  CREATE TABLE "movie_akas" (\n"movie_id" TEXT,\...  
4  CREATE TABLE "movie_ratings" (\n"movie_id" TEX...  
5  CREATE TABLE "persons" (\n"person_id" TEXT,\n ...  
6  CREATE TABLE "principals" (\n"movie_id" TEXT,\...  
7  CREATE TABLE "writers" (\n"movie_id" TEXT,\n  ...

Then we are going to take a peek into the database's ERD, to understand how the tables relate to each other.

![movie data erd](https://raw.githubusercontent.com/learn-co-curriculum/dsc-phase-2-project-v3/main/movie_data_erd.jpeg)


In [33]:
# This code queries the database to access the table

movie_basics = pd.read_sql("""SELECT *
                               FROM movie_basics""", conn)
movie_basics

movie_id                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                  None  
146143              NaN           Documentary  

[146144 rows x 6 columns]

In [34]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = movie_basics.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

runtime_minutes    31739
genres              5408
original_title        21
dtype: int64

In [35]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = movie_basics.duplicated().sum()
duplicates

0

In [36]:
# This code queries the database to access the table

directors = pd.read_sql("""SELECT *
                           FROM directors""", conn)
directors

movie_id   person_id
0       tt0285252   nm0899854
1       tt0462036   nm1940585
2       tt0835418   nm0151540
3       tt0835418   nm0151540
4       tt0878654   nm0089502
...           ...         ...
291169  tt8999974  nm10122357
291170  tt9001390   nm6711477
291171  tt9001494  nm10123242
291172  tt9001494  nm10123248
291173  tt9004986   nm4993825

[291174 rows x 2 columns]

In [37]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = directors.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

Series([], dtype: int64)

In [38]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = directors.duplicated().sum()
duplicates

127639

In [39]:
# This code queries the database to access the table

known_for = pd.read_sql("""SELECT *
                           FROM known_for""", conn)
known_for

person_id   movie_id
0        nm0061671  tt0837562
1        nm0061671  tt2398241
2        nm0061671  tt0844471
3        nm0061671  tt0118553
4        nm0061865  tt0896534
...            ...        ...
1638255  nm9990690  tt9090932
1638256  nm9990690  tt8737130
1638257  nm9991320  tt8734436
1638258  nm9991320  tt9615610
1638259  nm9993380  tt8743182

[1638260 rows x 2 columns]

In [40]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = known_for.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

Series([], dtype: int64)

In [41]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = known_for.duplicated().sum()
duplicates

0

In [42]:
# This code queries the database to access the table

movie_akas = pd.read_sql("""SELECT *
                            FROM movie_akas""", conn)
movie_akas

movie_id  ordering                                    title region  \
0       tt0369610        10                            Джурасик свят     BG   
1       tt0369610        11                        Jurashikku warudo     JP   
2       tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3       tt0369610        13                  O Mundo dos Dinossauros     BR   
4       tt0369610        14                           Jurassic World     FR   
...           ...       ...                                      ...    ...   
331698  tt9827784         2                       Sayonara kuchibiru   None   
331699  tt9827784         3                            Farewell Song    XWW   
331700  tt9880178         1                              La atención   None   
331701  tt9880178         2                              La atención     ES   
331702  tt9880178         3                            The Attention    XWW   

       language        types   attributes  is_original_title  
0            bg         None         None                0.0  
1          None  imdbDisplay         None                0.0  
2          None  imdbDisplay         None                0.0  
3          None         None  short title                0.0  
4          None  imdbDisplay         None                0.0  
...         ...          ...          ...                ...  
331698     None     original         None                1.0  
331699       en  imdbDisplay         None                0.0  
331700     None     original         None                1.0  
331701     None         None         None                0.0  
331702       en  imdbDisplay         None                0.0  

[331703 rows x 8 columns]

In [43]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = movie_akas.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

attributes           316778
language             289988
types                163256
region                53293
is_original_title        25
dtype: int64

In [44]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = movie_akas.duplicated().sum()
duplicates

0

In [45]:
# This code queries the database to access the table

movie_ratings = pd.read_sql("""SELECT *
                               FROM movie_ratings""", conn)
movie_ratings

movie_id  averagerating  numvotes
0      tt10356526            8.3        31
1      tt10384606            8.9       559
2       tt1042974            6.4        20
3       tt1043726            4.2     50352
4       tt1060240            6.5        21
...           ...            ...       ...
73851   tt9805820            8.1        25
73852   tt9844256            7.5        24
73853   tt9851050            4.7        14
73854   tt9886934            7.0         5
73855   tt9894098            6.3       128

[73856 rows x 3 columns]

In [46]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = movie_ratings.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

Series([], dtype: int64)

In [47]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = movie_ratings.duplicated().sum()
duplicates

0

In [48]:
# This code queries the database to access the table

persons = pd.read_sql("""SELECT *
                         FROM persons""", conn)
persons

person_id         primary_name  birth_year  death_year  \
0       nm0061671    Mary Ellen Bauder         NaN         NaN   
1       nm0061865         Joseph Bauer         NaN         NaN   
2       nm0062070           Bruce Baum         NaN         NaN   
3       nm0062195         Axel Baumann         NaN         NaN   
4       nm0062798          Pete Baxter         NaN         NaN   
...           ...                  ...         ...         ...   
606643  nm9990381         Susan Grobes         NaN         NaN   
606644  nm9990690          Joo Yeon So         NaN         NaN   
606645  nm9991320       Madeline Smith         NaN         NaN   
606646  nm9991786  Michelle Modigliani         NaN         NaN   
606647  nm9993380       Pegasus Envoyé         NaN         NaN   

                                      primary_profession  
0              miscellaneous,production_manager,producer  
1             composer,music_department,sound_department  
2                             miscellaneous,actor,writer  
3       camera_department,cinematographer,art_department  
4       production_designer,art_department,set_decorator  
...                                                  ...  
606643                                           actress  
606644                                           actress  
606645                                           actress  
606646                                          producer  
606647                             director,actor,writer  

[606648 rows x 5 columns]

In [49]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = persons.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

death_year            599865
birth_year            523912
primary_profession     51340
dtype: int64

In [50]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = persons.duplicated().sum()
duplicates

0

In [51]:
# This code queries the database to access the table

principals = pd.read_sql("""SELECT *
                            FROM principals""", conn)
principals

movie_id  ordering   person_id  category       job  \
0        tt0111414         1   nm0246005     actor      None   
1        tt0111414         2   nm0398271  director      None   
2        tt0111414         3   nm3739909  producer  producer   
3        tt0323808        10   nm0059247    editor      None   
4        tt0323808         1   nm3579312   actress      None   
...            ...       ...         ...       ...       ...   
1028181  tt9692684         1   nm0186469     actor      None   
1028182  tt9692684         2   nm4929530      self      None   
1028183  tt9692684         3  nm10441594  director      None   
1028184  tt9692684         4   nm6009913    writer    writer   
1028185  tt9692684         5  nm10441595  producer  producer   

                   characters  
0                 ["The Man"]  
1                        None  
2                        None  
3                        None  
4            ["Beth Boothby"]  
...                       ...  
1028181  ["Ebenezer Scrooge"]  
1028182   ["Herself","Regan"]  
1028183                  None  
1028184                  None  
1028185                  None  

[1028186 rows x 6 columns]

In [52]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = principals.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

job           850502
characters    634826
dtype: int64

In [53]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = principals.duplicated().sum()
duplicates

0

In [54]:
# This code queries the database to access the table

writers = pd.read_sql("""SELECT *
                         FROM writers""", conn)
writers

movie_id   person_id
0       tt0285252   nm0899854
1       tt0438973   nm0175726
2       tt0438973   nm1802864
3       tt0462036   nm1940585
4       tt0835418   nm0310087
...           ...         ...
255868  tt8999892  nm10122246
255869  tt8999974  nm10122357
255870  tt9001390   nm6711477
255871  tt9004986   nm4993825
255872  tt9010172   nm8352242

[255873 rows x 2 columns]

In [55]:
#The info() function gives us a summary of the data, but to check missing values closely, we are using .isnull() function to look at the count of missing vakues in each column.
# This code checks for null values in each column and sums them up. It then filters out columns with no missing values and returns those with null values.

missing = writers.isnull().sum()
missing[missing > 0].sort_values(ascending=False)

Series([], dtype: int64)

In [56]:
# This code counts the amount of duplicated records exist in the table.
# We are using the .dupiicated() function to identify the duplicated records and the .sum() function to count them.

duplicates = writers.duplicated().sum()
duplicates

77521

### Data summary
* The dataset contains structured data stored in a database containing 8 tables namely, 'movie_basics', 'directors', 'known_for', 'movie_akas', 'movie_ratings', 'persons', 'principals', and 'writers'.

### Table exploration
1. `movie_basics` - This table contains general movie information. It contains 146144 rows and 6 columns. Out of the 6 columns, 3 contain null values.

2. `directors` - This table links movie directors and the movies they directed. The data is stored in 291174 rows and 2 columns. It also contains 127639 duplicate records.

3. `known_for` - This table links people in the film industry and the movies they are known for. It has records in 1638260 rows and 2 columns and has no null values and duplicate records.

4. `movie_akas` - This table has information on the different names of the movies worldwide. It has 331703 rows and 8 columns. 5 out of the 8 columns contain null values.

5. `movie_ratings` - This table has information on movie ratings and number of votes. It contains 73856 rows and 3 columns, and has no null values or duplicated records.

6. `persons` - This table contains the biographies of all persons in the film industry including actors and directors. It has 606648 rows and 5 columns. Ou of the 5 columns, 3 contain null values.

7. `principals` - This table contains information about the job categories of the individuals in the film categories and for some, the characters they have played in past movies. It stored data in 1028186 rows and 6 columns. It has null values in 2 columns and no duplicate records.

8. `writers` - This table links nmovie writers and the movies they wrote. It contains 255873 rows and 2 columns. It also has 77521 duplicated records.

### Data Quality Observations
Most tables have a clean schema with clear foreign key references. However, it contains incomplete data and duplicated records.
- Joins between tables like movie_basics, movie_ratings, and persons are feasible and useful for analysis.

### Next Steps / Ideas for Deeper Analysis
* Join movie_basics with movie_ratings to find top-rated movies.

* Identify the most prolific directors or actors by linking directors, principals, and persons.

* Investigate trends by release year, genre, and average rating.

* Clean or supplement the known_for table if necessary.